### ISSUE TITLE
Printer support

### ISSUE THREAD

**Issue by cedrozor** [cedrozor/myrtille#4]
Myrtille doesn't support printer. In a previous implementation of it, Myrtille had a "print to PDF" feature (using PDFCreator), followed by the download of the generated PDF file.

I removed this feature when I extracted the core because it implied the installation of PDFCreator on the RDP server(s). Currently, nothing is required on them.

If you have another approach, please don't hesitate :)

**Comment by p33kabo0** [comment]
Hello there,

I know your request is closed, but if I read it correctly, if we want to use it, we can implement it with PDFCreator?  Have you remove the code for doing so?

That is my main concern about using Myrtille is for PDF printing. Do you have any documentation how I can implant it into my version?

Thank you
Philippe

**Comment by cedrozor** [comment]
Hi,

The previous implementation I spoke about was a PoC long time ago (2008), using PDFCreator.

I closed this request because I lack some time to implement it and I hoped some help that never came :/

But it's not so hard to do (compared to what I already done), so I may reconsider doing it, with something better (PDFCreator is now outdated and even contains some ads).

**Comment by cedrozor** [comment]
Yes, I know about Guacamole and have experienced their print feature.

Indeed, they use ghostscript and c code to handle it ([here](https://github.com/glyptodon/guacamole-server/blob/04205a9b979077580f2e2a10746797a6bab4c4e7/src/protocols/rdp/rdp_print_job.c#L41)), and stream it to the web client.

We had a different workflow with the PoC; PDFCreator had a "after print" event and allowed to define an action on it. So we passed the pdf filename to that action, retrieved and streamed it to the web client.

I can get some inspiration from Guacamole, but it's still work (I just lack time).

**Comment by p33kabo0** [comment]
Well, I will try to get the project in my visual studio and see if I could
understand your code first.

I'm more vb.net than c# but I can code a little in c# ...

I will test out all myrtille if it can really replace our standard rod
connection to be way more compatible and in windows form.

The thing I like about Myrtille is that it can be installed directly on my
server so its easier to manage users and user folders for sharing files.

If you can give me the guideline where to check in myrtille and I will take
a look.

Thank you
Philippe

**Comment by cedrozor** [comment]
You can follow the guideline here to build and debug myrtille: https://github.com/cedrozor/myrtille/blob/master/DOCUMENTATION.md#build (I used VS2015 but it's the same for VS2017)

**Comment by welsonplay** [comment]
I know you do not have this question that does not refer directly to Myrtille. Myrtille is part of a solution I'm selling to a customer. If an integration between Myrtille and PDFCreator is implemented, will I still be able to commit this solution?

Has anyone had any other solution for printing files where I do not have access to the printer in my environment?

**Comment by cedrozor** [comment]
You can do it on a fork then make a pull request, but I'd rather use ghostscript directly than relying on a third party. Also, PDFCreatror seems to be (or bring) some kind of adware from what I had read recently.
